lib/java/com/google/android/material/drawable/DrawableUtils.java

```diff
@@ -207,44 +230,52 @@ public static Drawable compositeTwoLayeredDrawable(
     }
     LayerDrawable drawable =
         new LayerDrawable(new Drawable[] {bottomLayerDrawable, topLayerDrawable});
+
+    final int topLayerOriginalWidth = topLayerDrawable.getIntrinsicWidth();
+    final int topLayerOriginalHeight = topLayerDrawable.getIntrinsicHeight();
+    final int bottomLayerOriginalWidth = bottomLayerDrawable.getIntrinsicWidth();
+    final int bottomLayerOriginalHeight = bottomLayerDrawable.getIntrinsicHeight();
+
     int topLayerNewWidth;
     int topLayerNewHeight;
-    if (topLayerDrawable.getIntrinsicWidth() == -1 || topLayerDrawable.getIntrinsicHeight() == -1) {
-      // If there's no intrinsic width or height, keep bottom layer's size.
-      topLayerNewWidth = bottomLayerDrawable.getIntrinsicWidth();
-      topLayerNewHeight = bottomLayerDrawable.getIntrinsicHeight();
-    } else if (topLayerDrawable.getIntrinsicWidth() <= bottomLayerDrawable.getIntrinsicWidth()
-        && topLayerDrawable.getIntrinsicHeight() <= bottomLayerDrawable.getIntrinsicHeight()) {
-      // If the top layer is smaller than the bottom layer in both its width and height, keep top
-      // layer's size.
-      topLayerNewWidth = topLayerDrawable.getIntrinsicWidth();
-      topLayerNewHeight = topLayerDrawable.getIntrinsicHeight();
+    if (topLayerDesiredWidth != -1 && topLayerDesiredHeight != -1) {
```

Comment:
I think we can just adjust width & height separately like:

topLayerNewWidth = topLayerDesiredWidth == -1 ? topLayerOriginalWidth : topLayerDesiredWidth
topLayerNewHeight = topLayerDesiredHeight == -1 ? topLayerOriginalHeight : topLayerDesiredHeight

And then fall back to the original logic.

Reply:
I suppose we could. But I decided to fix some edge cases, that's why there are a little more changes than expected.

Reply:
I guess a more important question here is if we want to handle width and height separately?

Like, can we only specify a desired width but not a desired height?

Reply:
Ah, got it. Well, I can add this, why not.